hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/ByteArrayObjectDataInput.java

```diff
@@ -535,7 +603,9 @@ public final Object readObject() throws IOException {
 
     @Override
     public final Data readData() throws IOException {
-        return service.readData(this);
+        byte[] bytes = readByteArray();
+        Data data = bytes != null ? new HeapData(bytes) : null;
```

Comment:
Flavor thing; I always try to get rid of !. 

Data data = bytes == null ? null : new HeapData(bytes);

Makes it easier to read because I don't need to flip the expression.